### PR TITLE
[SPARK-36499][SQL][TESTS] Test Interval multiply / divide null

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

Test the following valid operations:
```
year-month interval * null
null * year-month interval
year-month interval / null
```
and invalid operations:
```
null / interval
int / interval
```

### Why are the changes needed?

Improve test coverage

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Pass CI

### CODE REVIEW

File: sql/core/src/test/resources/sql-tests/inputs/interval.sql [hunk 1]
```diff
@@ -170,6 +170,21 @@ select
   null - interval '2' year,
   null + interval '2' hour,
   null - interval '2' hour;
+select
```
Comment (cloud-fan) [inline sql/core/src/test/resources/sql-tests/inputs/interval.sql:173]: there is a section at the beginning of this file `-- multiply and divide an interval by a number`. Can we put the null tests there?

Comment (gengliangwang) [reply sql/core/src/test/resources/sql-tests/inputs/interval.sql]: Done

**Review by cloud-fan — APPROVED** [review]

File: sql/core/src/test/resources/sql-tests/results/ansi/interval.sql.out [hunk 2]
```diff
@@ -117,6 +117,75 @@ struct<(INTERVAL '02' SECOND * NULL):interval day to second>
...
+struct<>
+-- !query output
+java.lang.ArithmeticException
+/ by zero
+
+
+-- !query
+select interval '2 year' / null
+-- !query schema
+struct<(INTERVAL '2' YEAR / NULL):interval year to month>
```
Comment (MaxGekk) [inline sql/core/src/test/resources/sql-tests/results/ansi/interval.sql.out:132]: I wonder why the type is `interval year to month` but not `interval year`

File: sql/core/src/test/resources/sql-tests/inputs/interval.sql [hunk 3]
```diff
@@ -16,6 +16,16 @@ select interval '2 seconds' / 0;
 select interval '2 seconds' / null;
 select interval '2 seconds' * null;
 select null * interval '2 seconds';
+select interval '2 year' / 0;
+select interval '2 year' / null;
+select interval '2 year' * null;
+select null * interval '2 year';
```
Comment (MaxGekk) [inline sql/core/src/test/resources/sql-tests/inputs/interval.sql:22]: Can you check interval in ANSI style at least once: `interval '2' year`

File: sql/core/src/test/resources/sql-tests/results/ansi/interval.sql.out (hunk 2 above)
Comment (cloud-fan) [reply sql/core/src/test/resources/sql-tests/results/ansi/interval.sql.out:132]: good point. This is probably something we should fix.

Comment (MaxGekk) [reply sql/core/src/test/resources/sql-tests/results/ansi/interval.sql.out:132]: I am ok to fix this in a separate PR. @gengliangwang Could you open a sub-task in https://issues.apache.org/jira/browse/SPARK-27790, please.

File: sql/core/src/test/resources/sql-tests/inputs/interval.sql (hunk 3 above)
Comment (gengliangwang) [reply sql/core/src/test/resources/sql-tests/inputs/interval.sql:22]: Done, updated.

**Review by MaxGekk — APPROVED** [review]

File: sql/core/src/test/resources/sql-tests/results/ansi/interval.sql.out (hunk 2 above)
Comment (gengliangwang) [reply sql/core/src/test/resources/sql-tests/results/ansi/interval.sql.out:132]: This is the data type of `DivideYMInterval`, which is the default YM interval type.

Comment (gengliangwang) [reply sql/core/src/test/resources/sql-tests/results/ansi/interval.sql.out:132]: It's to have a wider data type as the division result. Just like the following
```
scala> spark.sql("select 1 /null")
res0: org.apache.spark.sql.DataFrame = [(1 / NULL): double]
```

Comment (MaxGekk) [reply sql/core/src/test/resources/sql-tests/results/ansi/interval.sql.out:132]: Got it. Thanks.

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master/3.2.
Thank you, @gengliangwang and @cloud-fan for review.

**Pull request closed without merging**